scalableminds/webknossos
# Don't display "Fill segment" option if editable mapping is active

### URL of deployed dev instance (used for testing):
- https://disablefillsegmentwithactiveeditablemapping.webknossos.xyz/
- e.g. DS ROI2017_wkw_sparse_segmentation (I think it only has JSON mappings, not sure where a HDF5 mapping can be found to reproduce the bug)

### How to test
- create editable mapping and make sure that flood filling is not available in the context menu any more
- advanced: change code so that it is available in context menu again, and click is while editable mapping is active. there should be a toast and a console.error.
- in an annotation with a segmentation layer, turn it invisible and turn on a skeleton layer. now only skeleton options should be available in the context menu in the viewport, matching the disabled volume actions in the action bar.

### TODOs:
- [x] disable flood filling if there is no active segmentation layer
- [x] see if disabledVolumeInfo can be used even more
- [x] DRY nonSkeletonActions

### Issues:
- fixes #7925 

------
(Please delete unneeded items, merge only when none are left open)
